### PR TITLE
Semantic implementation for mid-project report

### PR DESCRIPTION
**Pull request by hextraza**
If u know u know

### CODE REVIEW

File: Main.hs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+-- Otherwise, throw an error.
+listExprUn :: ListOpUn -> Expr -> Expr
+listExprUn (Ind i) (List v) = v !! i
+listExprUn (Map n) (List v) = undefined
+listExprUn _ _ = Error "Cannot map or index non-list type."
+
+-- Takes an expression; if it's a list, perform the given operation on it.
+-- Otherwise, throw an error.
+listExprBi :: ListOpBi -> Expr -> Expr -> Expr
+listExprBi ListConcat (List r) (List l) = List $ r ++ l
```
Comment (philectron) [inline Main.hs]: Is it left-right or right-left?

File: Main.hs [hunk 2]
```diff
@@ -79,7 +79,9 @@ boolExprBi _ _ _ = Error "Cannot perform binary boolean operation on non-strings
 -- Takes an expression; if it's a list, perform the given operation on it.
 -- Otherwise, throw an error.
 listExprUn :: ListOpUn -> Expr -> Expr
-listExprUn = undefined
+listExprUn (Ind i) (List v) = v !! i
+listExprUn (Map n) (List v) = undefined
```
Comment (philectron) [inline Main.hs:81]: Mapping is not done yet, but I think this functionality is a soft requirement among the list operations required to implement.

File: Main.hs [hunk 3]
```diff
@@ -55,9 +55,7 @@ arithExpr _ _ _ = Error "Cannot perform arithmetic operation on non-integers"
 -- bool-un-op on it and returns the result.
 -- Otherwise, throws an error.
 boolExprUn :: BoolOpUn -> Expr -> Expr
-boolExprUn op (B bool) = B $ operator bool
-  where operator = case op of
-                     Not -> (not)
+boolExprUn Not (B bool) = B $ not bool
```
Comment (philectron) [inline Main.hs:58]: 😠

**Review by philectron — COMMENTED** [review]
If you don't know, now you know.

File: Main.hs (hunk 2 above)
Comment (hextraza) [reply Main.hs:81]: Yeah, but it shouldn't be required for the 50% mark. Mapping will require completed functions, which is not a part of the 50%.

File: Main.hs (hunk 3 above)
Comment (hextraza) [reply Main.hs:58]: got get

File: Main.hs (hunk 1 above)
Comment (hextraza) [reply Main.hs]: Corrected.

File: Main.hs [hunk 4]
```diff
@@ -84,7 +84,7 @@ listExprUn _ _ = Error "Cannot map or index non-list type."
 -- Takes an expression; if it's a list, perform the given operation on it.
 -- Otherwise, throw an error.
 listExprBi :: ListOpBi -> Expr -> Expr -> Expr
-listExprBi ListConcat (List r) (List l) = List $ r ++ l
+listExprBi ListConcat (List l) (List r) = List $ l ++ r
```
Comment (philectron) [inline Main.hs:87]: image

**Review by philectron — APPROVED** [review]

**Review by khashf — APPROVED** [review]

**Pull request merged**